src/lib/dai.c

```diff
@@ -134,6 +135,50 @@ static inline const struct dai_type_info *dai_find_type(uint32_t type)
 	return NULL;
 }
 
+#ifdef __ZEPHYR__
+struct dai *dai_get_zephyr(const struct dai_type_info *dti, uint32_t type, uint32_t index,
+			   uint32_t flags)
+{
+	struct dai *d;
+	int ret;
+
+	/* currently only ssp supported, add more when ported to zephyr */
+	if (type == SOF_DAI_INTEL_SSP) {
+		tr_info(&dai_tr,"log_ssp_data get_dai_zephyr index: %u", index);
+		for (d = dti->dai_array; d < dti->dai_array + dti->num_dais; d++) {
+			if (d->index != index)
+				continue;
+
+			if (flags & DAI_CREAT)
+				tr_info(&dai_tr, "dai_get_zephyr CREATE %u", d->index);
+			else
+				tr_info(&dai_tr, "dai_get_zephyr no CREATE flag%u", d->index);
+
+			ret = dai_probe_op(d);
```

Comment:
for other DAIs `dai_probe_op()` is only called if the `CREAT` flag is set and if `sref == 0`, don't we need to do the same here or is this checked internally by Zephyr's SSP probing?

Reply:
yeah so I moved that to driver. I have to say I'm not totally sure how this should work. Now the driver has internal knowledge if it has been "created" i.e. its internal memory has been allocated. if it has been allocated, only ref count is increased. And vice versa in remove. So this should work without the create flag, as the functionality is creation and possible ref counting. So don't see a use case where we would have ref count increase before creation, as the creation/probe happens in sof init. But if someone has better idea, I'm all ears...